docs/user/http-redirect.md

```diff
@@ -0,0 +1,117 @@
+# HTTP Redirects
+
+The [HTTPRoute][] resource can issue redirects to clients or rewrite paths sent upstream using filters. Note that
+HTTPRoute rules cannot use both filter types at once. Currently, Envoy Gateway only supports core [HTTPRoute filters][]
+which consist of `RequestRedirect` and `RequestHeaderModifier`. To learn more about HTTP routing, refer to the
+[Gateway API documentation][].
+
+Follow the steps from the [Secure Gateways](SECURE_GATEWAY.md) to install Envoy Gateway and the example manifest. Do not
+proceed until you can curl the example backend from the Quickstart guide using HTTPS.
+
+## Redirects
+Redirects return HTTP 3XX responses to a client, instructing it to retrieve a different resource. A
+[`RequestRedirect` filter][req_filter] instructs Gateways to emit a redirect response to requests that match the rule.
+For example, to issue a permanent redirect (301) from HTTP to HTTPS, configure `requestRedirect.statusCode=301` and
+`requestRedirect.scheme="https"`:
+
+```shell
+cat <<EOF | kubectl apply -f -
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: http-to-https-filter-redirect
+spec:
+  parentRefs:
+    - name: eg
+  hostnames:
+    - redirect.example
+  rules:
+    - filters:
+      - type: RequestRedirect
+        requestRedirect:
+          scheme: https
+          statusCode: 301
+          hostname: www.example.com
+          port: 8443
+      backendRefs:
+      - name: httpbin
+        port: 80
+EOF
+```
+
+__Note:__ `301` (default) and `302` are the only supported statusCodes.
+
+The HTTPRoute status should indicate that it has been accepted and is bound to the example Gateway.
+
+```shell
+kubectl get httproute/http-to-https-filter-redirect -o yaml
+```
+
+Curl'ing `redirect.example/get` should result in a `301` response from the example Gateway and redirecting to the
+configured redirect hostname.
+
+```shell
+$ curl -L -vvv --header "Host: redirect.example" "http://${GATEWAY_HOST}:8080/get"
```

Comment:
DO we need to let user know how to configure GATEWAY_HOST?

like 
```
export GATEWAY_HOST=$(kubectl get gateway/eg -o jsonpath='{.status.addresses[0].value}')
```

Reply:
I don't think its totally necessary, but I think its a good idea. I think one of the assumptions we make for all these specific docs pages is that the user has already gone through the quickstart, so they should know how to get the address, but eliminating friction anywhere we can sounds like a good idea to me.

Reply:
This doc does state to complete the Quickstart but it doesn't hurt to add the export. I'll update and push a new commit.

Reply:
Resolved feedback in 97e26df.